chain/events/filter/index.go

```diff
@@ -356,6 +356,16 @@ func (ei *EventIndex) migrateToVersion4(ctx context.Context) error {
 		}
 	}
 
+	stmtEventIndexUpdate, err := tx.PrepareContext(ctx, "UPDATE event SET event_index = (SELECT COUNT(*) FROM event e2 WHERE e2.tipset_key_cid = event.tipset_key_cid AND e2.id <= event.id) - 1")
```

Comment:
I'm a bit worried about bundling this into the same migration. I'd be inclined to just introduce a v5 and stack them up just to be safe for anyone running on `master`. If we always move `ei.vacuumDBAndCheckpointWAL(ctx)` to the very last migration so we only do it once then the cost is pretty low.

Maybe this one isn't so critical but we've found from experience that people run on `master` and we don't have a good way of dealing with these migrations.

Reply:
@rvagg I agree that migrating to version 5 makes more sense here.